src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -56,20 +61,52 @@ class CommonController extends FrameworkBundleAdminController
      */
     public function notificationsAction()
     {
-        // TODO: Use CQRS
-        return new JsonResponse((new \Notification())->getLastElements());
+        $employeeId = Context::getContext()->employee->id;
+        /** @var NotificationsResults $elements */
+        $elements = $this->getQueryBus()->handle(new GetNotificationLastElements($employeeId));
+
+        $response = [];
+
+        foreach ($elements->getNotificationsResults() as $element) {
```

Comment:
I'm wondering 🤔 ... could we make this simpler ?

Some ideas:
- implements a `toArray()` function into the DTO NotificationsResult, NotificationsResults and NotificationResult
-  what happens if we use Symfony Serializer here ? I think he might be able to perform this operation without the need for us to write it (see below)

I'm not sure this is a good idea nor it's working 😅 please tell me what you think

Serializer usage:
```
// should work, not 100% sure, might need a few fix
$serializer = $this->get('serializer');
$notifications[] = $serializer->serialize(notification);
```